mava/systems/q_learning/rec_iql.py

```diff
@@ -564,33 +564,36 @@ def run_experiment(cfg: DictConfig) -> float:
         )
 
     max_episode_return = -jnp.inf
-    start_time = time.time()
 
     # Main loop:
     for eval_idx, t in enumerate(
-        range(steps_per_rollout, int(cfg.system.total_timesteps), steps_per_rollout)
+        range(steps_per_rollout, int(cfg.system.total_timesteps + 1), steps_per_rollout)
     ):
         # Learn loop:
+        start_time = time.time()
         learner_state, (metrics, losses) = update(learner_state)
         jax.block_until_ready(learner_state)
 
         # Log:
         # Add learn steps here because anakin steps per second is learn + act steps
         # But we also want to make sure we're counting env steps correctly so
         # learn steps is not included in the loop counter.
-        learn_steps = anakin_steps * cfg.system.epochs
-        sps = (t + learn_steps) / (time.time() - start_time)
+        elapsed_time = time.time() - start_time
         eps = jax.numpy.maximum(
             cfg.system.eps_min, 1 - (t / cfg.system.eps_decay) * (1 - cfg.system.eps_min)
         )
-
         final_metrics, ep_completed = episode_metrics.get_final_step_metrics(metrics)
+        final_metrics["steps_per_second"] = steps_per_rollout / elapsed_time
```

Comment:
So this is why I originally combined the two steps per second values. It's not fair to say that we do `steps_per_rollout / elapsed_time` because we aren't only doing `steps_per_rollout` actor steps here, we are also doing `anakin_steps * cfg.system.epochs` gradient steps here. 

So it's not really a true representation of either the learner steps per second or the actor steps per second.

Could you maybe put this in a comment above or make a new issue to fix this in all the all the systems :pray:

Reply:
For sure, I will make an issue!